File: bld/java-basic/src/main/java/com/example/JokeLoader.java
package com.example;

import java.io.IOException;
import java.util.List;

public class JokeLoader {

    private final FileReader fileReader = new FileReader();

    public List<String> loadJoke(int id) {
        try {
            return fileReader.readFile("joke_" + id + ".txt");
        } catch (IOException e) {
            throw new RuntimeException("Joke with id " + id + " not found!", e);
        }
    }

    public int getNumberOfJokes() {
        return 3;
    }
}

File: _template/kotlin-basic/src/main/kotlin/com/example/JokeLoader.kt
package com.example

class JokeLoader {
    private val fileReader = FileReader()

    fun loadJoke(id: Int): List<String> {
        return fileReader.readFile("joke_$id.txt")
            ?: error("Joke with id $id not found!")
    }

    companion object {
        const val NUMBER_OF_JOKES = 3
    }
}
